### PR TITLE
add option for everything search in full path (default true)

### LINKED ISSUE #1894: [Enhancement] 程序插件的文件夹结果排序
**Is your feature request related to a problem? Please describe.**

我想要通过直接输入文件夹名称来打开我想要的文件夹：
观察到【程序】插件会按照名称排序，导致此Z1A的文件夹排名靠后，在文件夹结构复杂的时候，甚至有可能找不到我想要的这个文件夹。
如图当我输入Z1A的时候，我希望默认选项应该是Z1A本身而不是它的子文件夹。
截图_20230210110756
> I want to open the folder I want by directly typing the folder name:
> It is observed that the [Program] plug-in will be sorted the result by alphabetical order, resulting in the lower ranking of the Z1A folder. When the folder structure is complicated, it may even not be possible to find the folder I want.
> As shown in the picture, when I enter Z1A, I hope the default option should be Z1A itself instead of its subfolders.

**Describe the solution you'd like**

将everything返回的列表重新二级排序，第一层是路径长度排序，第二层是文件夹名称排序，这样就能快速且符合逻辑地找到我想要的文件夹。
> Re-sort the list returned by everything. Sort by path length first, then by name, so that I can quickly and logically find the folder I want.

**Describe alternatives you've considered**

或者将全字匹配的返回项作为默认项。
> Or simply make whole word match first

**Additional context**

...

### PR DESCRIPTION
**Pull request by taooceros**
fix #1894 

Tested:
- [x] Option works

### CODE REVIEW

**Comment by jjw24** [comment]
What about making the default as false? When true alot of subfolders with matching paths come out that confuses users not familiar with this option.

**Comment by taooceros** [comment]
ok makes sense

File: Plugins/Flow.Launcher.Plugin.Explorer/Languages/en.xaml [hunk 1]
```diff
@@ -127,6 +127,8 @@
     <system:String x:Key="flowlauncher_plugin_everything_sort_by_descending">&#x2193;</system:String>
     <system:String x:Key="flowlauncher_plugin_everything_nonfastsort_warning">Warning: This is not a Fast Sort option, searches may be slow</system:String>
 
+    <system:String x:Key="flowlauncher_plugin_everything_search_fullpath">Search In Full Path</system:String>
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.Explorer/Languages/en.xaml:130]:
```suggestion
    <system:String x:Key="flowlauncher_plugin_everything_search_fullpath">Search Full Path</system:String>
```

**Review by jjw24 — APPROVED** [review]

**Comment by VictoriousRaptor** [comment]
Just found that it breaks regex searches. if I input `wsa*.7z` when it's true it can't match files like `wsa_1254798tihdsafgoih.7z`.

**Comment by jjw24** [comment]
Let's issue this if you haven't already :)

**Pull request merged**